packages/hoppscotch-common/src/services/tab/rest.ts

```diff
@@ -0,0 +1,51 @@
+import { isEqual } from "lodash-es"
+import { getDefaultRESTRequest } from "~/helpers/rest/default"
+import { HoppRESTDocument, HoppRESTSaveContext } from "~/helpers/rest/document"
+import { TabService } from "./tab"
+
+export class RESTTabService extends TabService<HoppRESTDocument> {
+  public static readonly ID = "REST_TAB_SERVICE"
+
+  constructor() {
+    super()
+
+    this.tabMap.set("test", {
+      id: "test",
+      document: {
+        request: getDefaultRESTRequest(),
+        isDirty: false,
+        optionTabPreference: "params",
+      },
+    })
+
+    this.watchCurrentTabID()
+  }
+
+  public getTabRefWithSaveContext(ctx: HoppRESTSaveContext) {
+    for (const tab of this.tabMap.values()) {
+      // For `team-collection` request id can be considered unique
+      if (ctx && ctx.originLocation === "team-collection") {
+        if (
+          tab.document.saveContext?.originLocation === "team-collection" &&
+          tab.document.saveContext.requestID === ctx.requestID
+        ) {
+          return this.getTabRef(tab.id)
+        }
+      } else if (isEqual(ctx, tab.document.saveContext)) {
+        return this.getTabRef(tab.id)
+      }
+    }
+
+    return null
+  }
+
+  public getDirtyTabsCount() {
+    let count = 0
+
+    for (const tab of this.tabMap.values()) {
+      if (tab.document.isDirty) count++
+    }
+
+    return count
+  }
```

Comment:
Is it possible to abstract the common parts into helpers or something along those lines since `GraphQL` also uses a similar implementation?

Applicable to the `getTabRefWithSaveContext` method as well.

Reply:
`saveContext` is different for each `TabService`. so we need to keep that out from the abstract `TabService`

Reply:
Each tab service has its own `dirty` mechanism. Also, this is inside a tab `document`. We don't know the document's type.